source/common/listener_manager/listener_manager_impl.cc

```diff
@@ -642,6 +642,11 @@ absl::StatusOr<bool> ListenerManagerImpl::addOrUpdateListenerInternal(
 bool ListenerManagerImpl::hasListenerWithDuplicatedAddress(const ListenerList& list,
                                                            const ListenerImpl& listener) {
   for (const auto& existing_listener : list) {
+    // Skip if the listener is the same. It's because that the listener
+    // with same name was proven to be incompatible.
+    if (existing_listener->name() == listener.name()) {
+      continue;
+    }
```

Comment:
Could you please also add a test for this?

Reply:
I prefer to return false at the `hasDuplicatedAddress` to reduce the scope of code change.

Reply:
I think it's better to consolidate the name of the function `hasDuplicatedAddress()` to keep here.
And the effect of this code is very small because `hasListenerWithDuplicatedAddress()` is referred by `setNewOrDrainingSocketFactory()` only.